src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -33,6 +36,71 @@ public enum SrCnnDetectMode
         AnomalyAndExpectedValue = 2
     }
 
+    /// <summary>
+    /// The Deseasonality modes of SrCnn models. The de-seasonality mode is envoked when the period of the series is greater than 0.
+    /// </summary>
+    public enum SrCnnDeseasonalityMode
+    {
+        /// <summary>
+        /// In this mode, the stl decompose algorithm is used to de-seasonality.
+        /// </summary>
+        Stl = 0,
+
+        /// <summary>
+        /// In this mode, the mean value of points in the same position in a period is substracted to de-seasonality.
+        /// </summary>
+        Mean = 1,
+
+        /// <summary>
+        /// In this mode, the median value of points in the same position in a period is substracted to de-seasonality.
+        /// </summary>
+        Median = 2
+    }
+    public sealed class SrCnnEntireAnomalyDetectorOptions
+    {
+        [Argument(ArgumentType.Required, HelpText = "The name of the input column.", ShortName = "input",
+            SortOrder = 1, Purpose = SpecialPurpose.ColumnName)]
+        public string InputColumnName;
+
+        [Argument(ArgumentType.Required, HelpText = "The name of the output column.", ShortName = "output",
+            SortOrder = 2)]
+        public string OutputColumnName;
```

Comment:
Question to @yaeldekel and @harishsk : Is it ok to include the InputColumnName and OutputColumnName as part of the Options class? Since we _need_ the user to provide a valid name for both, and we can't provide a `Default` name for both, perhaps it's better to move these out of the Options class and actually add them as obligatory parameters on the public API?

Reply:
Yes, InputColumnName and OutputColumnName shouldn't be part of the Options class

Reply:
Refactored this class.